test-cypress/steps/home-steps.js

```diff
@@ -0,0 +1,122 @@
+class HomeSteps {
+
+    static tutorialVisible(shouldBeVisible) {
+        if (shouldBeVisible) {
+            cy.get('.tutorial-container')
+                .should('be.visible');
+        } else {
+            cy.get('.tutorial-container')
+                .should('not.be.visible');
+        }
+    }
+
+    static declineTutorial() {
+        // Click "No, thanks" button
+        cy.get('.tutorial-container .decline-tutorial').click();
+    }
+
+    static selectSPARQLQueryToExecute = function (query) {
+        cy.get('ul.saved-queries li:contains("' + query + '")').should("be.visible").trigger('hover');
+        // TODO maybe we can remove the force
+        cy.get('ul.saved-queries li:contains("' + query + '") .execute-saved-query').click({force: true});
+    };
+
+    static verifyQueryLink(queryName, modifiesRepoModal, queryURL) {
+        HomeSteps.selectSPARQLQueryToExecute(queryName);
+        modifiesRepoModal ? cy.get('.modal-body').should('be.visible') : cy.get('.modal-body').should('not.be.visible');
+        cy.url().should('eq', Cypress.config("baseUrl") + queryURL);
+        cy.visitAndWaitLoader('/');
+    }
+
+    static selectRepo(repositoryId) {
+        cy.visitAndWaitLoader('/');
+        cy.get('ul.repos')
+            .contains(repositoryId)
+            .should('be.visible')
+            .click();
+    }
+
+    static createRepo() {
+        let repositoryId = 'home-repository-' + Date.now();
+        cy.createRepository({id: repositoryId});
+        return repositoryId;
+    }
+
+    static isSelectedRepository(repositoryId) {
+        cy.get('ul.repos')
+            .should('contain', repositoryId);
+        cy.get('ul.repos')
+            .contains(repositoryId).and('be.hidden');
+        cy.get('#btnReposGroup')
+            .should('contain', repositoryId);
+    }
+
+    static hasRepositoryInfo(repositoryId) {
```

Comment:
These and other similar fragments with such selection duplications could be optimized like this. I havent tested it, just folowed the cypress API guides.
```javascript
cy.get('.active-repo-card').should('contain', repositoryId)
  .and('contain', 'total statements')
  .and('contain', 'explicit')
  .and('contain', 'inferred')
  .and('contain', 'expansion ratio')
  .within(() => {
      cy.get('.total-statements').should('contain', '70');
      cy.get('.explicit-statements').should('contain', '0')
      .and('contain', '70');
  });
```

Reply:
Yes, looks much better. I updated also isCorrectCreateRepositoryLink and selectSPARQLQueryToExecute to remote duplicate selectors.